Sources/HtmlPrettyPrint/Pretty.swift

```diff
@@ -33,34 +33,37 @@ private func prettyPrint(node: Node) -> Doc {
 private func prettyPrint(element: Element) -> Doc {
 
   return prettyPrintOpenTag(element: element)
-    <> (
-      element.content.map {
-        $0.map(prettyPrint(node:))
-          .vcat()
-          .indent(2)
-        } ?? .zero
-    )
+    <> prettyPrintChildren(nodes: element.content)
     <> prettyPrintCloseTag(element: element)
 }
 
+
+private func prettyPrintChildren(nodes: [Node]?) -> Doc {
+  guard let nodes = nodes else { return .empty }
+
+  return nodes.map(prettyPrint(node:))
+    .vcat()
+    .indent(2)
+}
+
 private func prettyPrintOpenTag(element: Element) -> Doc {
 
   return .text("<")
     <> .text(element.name)
     <> prettyPrint(attributes: element.attribs)
-    <> (element.content == nil ? .zero : .text(">") <> .hardline)
+    <> .text(">") <> (element.content == nil ? .empty : .hardline)
 }
 
 private func prettyPrintCloseTag(element: Element) -> Doc {
   return element.content == nil
-    ? .text(" />")
```

Comment:
turns out the valid way to do a "void" tag is just closing with `>` not `/>`, e.g. `<img src="">`